operations.go

```diff
@@ -6,154 +6,160 @@ import (
 	"math/big"
 )
 
-type Int big.Int
+type Int struct {
+	Value *big.Int
+	Base  *big.Int
+}
 
 // Default prime (256-bit secp256k1 EC order)
 var (
-	p, _ = IntFromString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141", 16)
+	defaultP, _ = new(big.Int).SetString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141", 16)
 )
 
-// SetP resets the (prime) modulus across the package
-func SetP(new_prime *Int) *Int {
-	p = new_prime
-	return p
-}
-
-func GetP() *Int {
-	return p
-}
-
 // ARITHMETIC OPERATIONS
 
 // Add - Modular addition of (an arbitrary number of) field elements
-func (n *Int) Add(nums ...*Int) *Int {
-	out := big.NewInt(0)
-	for _, n := range nums {
-		out.Add(out, (*big.Int)(n))
-	}
-	out.Mod(out, (*big.Int)(p))
-	*n = (Int)(*out)
+func (n *Int) Add(x, y *Int) *Int {
+	v := new(big.Int).Add(x.Value, y.Value)
+	n.Value = v.Mod(v, x.Base)
+	n.Base = x.Base
```

Comment:
This is similar to what I pointed at above: we are assigning pointers, so the underlying value is going to be the same in all instances. Is this what we desire?

Reply:
for primeBase I was thinking yes so we don't make tons of copies of the same prime. we should never be fiddling with the bases directly or else it's true, we could get weird behavior or data races.